tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServiceWrapperCommand.java

```diff
@@ -185,24 +207,20 @@ private void getServiceWrapperList( final List<String> wrapperList, String name
         }
     }
 
-    @SuppressWarnings( "unchecked" )
-    private String[] getStaticServiceWrapper() throws Exception
+    private String[] getServiceWrapperFromTargetPlatform() throws Exception
     {
-        final URL url =
-            FileLocator.toFileURL( ProjectCore.getDefault().getBundle().getEntry( "OSGI-INF/wrappers-static.json" ) );
-        final File servicesFile = new File( url.getFile() );
+        String[] result;
 
-        if( servicesFile.exists() )
+        if( _serviceWrapperName == null )
         {
-            final ObjectMapper mapper = new ObjectMapper();
-
-            List<String> map = mapper.readValue( servicesFile, List.class );
-            String[] wrappers = map.toArray( new String[0] );
-
-            return wrappers;
+            result = TargetPlatformUtil.getServiceWrapperList();
+        }
+        else
+        {
+            result = TargetPlatformUtil.getServiceWrapperBundle( _serviceWrapperName );
```

Comment:
this result is just String[2], so that is not good, we need to redesign these to be more semantic.